### PR TITLE
Remove the displayUnits from UnitManager

### PR DESCRIPTION
**Pull request by muybien3433**
According to  #1408

### CODE REVIEW

**Comment by bevans2000** [comment]
This unit tests and build are failing. The change is using UI components in the core package which cannot work. The code is built without the UI because we have another command line version which contains no UI code.
Get the new getDisplayUnits method to use the Unit manager instead of the core calling the ui

**Comment by mokun** [comment]
> The change is using UI components in the core package which cannot work.

Yeah that seems to be the culprit.

@muybien3433, 

For now, if you like, you can just add back the Set<Unit> displayUnits and addDisplayUnits() method the way it is and let the rest of the changes you made to come through those tests successfully.

Afterward, you can always make a separate pull to deal with the displayUnits issue.

**Comment by bevans2000** [comment]
> 
> @muybien3433,
> 
> For now, if you like, you can just add back the Set displayUnits and addDisplayUnits() method the way it is and let the rest of the changes you made to come through those tests successfully.
> 
Sounds a good plan. The correct solution is to fix the root cause of the concept of the display units. Means reworking the navigation tool logic but that's a different and bigger problem. So as @mokun says, put the display unit back in Unit manager as a short term.
We've got lots of time to close off 3.9 and clean it up

**Comment by muybien3433** [comment]
You're right, I'll leave it as it is for now.

File: mars-sim-core/src/main/java/com/mars_sim/core/UnitManager.java [hunk 1]
```diff
@@ -321,12 +320,11 @@ public synchronized void addUnit(Unit unit) {
 			switch (unit.getUnitType()) {
 				case SETTLEMENT -> {
 					lookupSettlement.put(unitIdentifier, (Settlement) unit);
-					navpointPanel.addDisplayUnit(unit);
+					addDisplayUnit(unit);
 				}
 				case PERSON -> lookupPerson.put(unitIdentifier, (Person) unit);
 				case ROBOT -> lookupRobot.put(unitIdentifier, (Robot) unit);
 				case VEHICLE -> lookupVehicle.put(unitIdentifier, (Vehicle) unit);
```
Comment (mokun) [inline mars-sim-core/src/main/java/com/mars_sim/core/UnitManager.java]: I suppose we still need to call addDisplayUnit(unit) here under case VEHICLE.

**Comment by mokun** [comment]
@muybien3433 

Currently, it still fails some maven tests for whatever reasons.

```
Error:  Errors: 
Error:    TestContainment.testBagOnSurface:214 » IllegalArgument Cannot store unit type:CONTAINER
Error:    TestContainment.testBagOnVehicle:231 » IllegalArgument Cannot store unit type:CONTAINER
Error:    TestSaving.testSaving:44 » IllegalArgument Cannot store unit type:CONTAINER
Error:    EquipmentInventoryTest.testAmountInEquipmentLoading:64 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadPersonTest.testAssignThermalBottle:89 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadPersonTest.testTransferEquipment:77 » IllegalArgument Cannot store unit type:CONTAINER
Error:    DigLocalTest.testCreateIceTask:65 » IllegalArgument Cannot store unit type:CONTAINER
Error:    DigLocalTest.testCreateRegolithTask:27 » IllegalArgument Cannot store unit type:CONTAINER
Error:    DigLocalTest.testCreateTaskWithBin:102 » IllegalArgument Cannot store unit type:CONTAINER
Error:    DigLocalTest.testIceMetaTask:127 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadControllerTest.testLoadFull:290->testLoading:299->loadSettlement:490->loadSettlementEquipment:331 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadControllerTest.testLoadMissingOptionalEquipment:146->loadSettlement:490->loadSettlementEquipment:331 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadControllerTest.testLoadOptionalEquipment:133->testLoading:299->loadSettlement:490->loadSettlementEquipment:331 » IllegalArgument Cannot store unit type:CONTAINER
Error:    LoadControllerTest.testLoadRequiredEquipment:119->testLoading:299->loadSettlement:490->loadSettlementEquipment:331 » IllegalArgument Cannot store unit type:CONTAINER
```

Q: Can you find TestContainment (under mars-sim-core package) and run this maven test in your IDE ? 

In TestContainment.java,
```java
	/*
	 * Test method for 'com.mars_sim.simulation.person.ai.task.LoadVehicle.isFullyLoaded()'
	 */
	public void testVehicleOnSurface() throws Exception {
		Vehicle vehicle = buildRover(settlement, "Garage Rover", new LocalPosition(1,1));

		assertTrue("Transfer to Mars surface but still within settlement vicinity", vehicle.transfer(surface));

		assertWithinSettlementVicinity("After transfer from Settlement", vehicle);
	}

	/*
	 * Test method for 'com.mars_sim.simulation.person.ai.task.LoadVehicle.isFullyLoaded()'
	 */
	public void testBagOnSurface() throws Exception {

		Equipment bag = EquipmentFactory.createEquipment(EquipmentType.BAG, settlement);
		
		assertInsideSettlement("Initial equipment", bag, settlement);
		
		assertTrue("Transfer to Mars surface but still within settlement vicinity", bag.transfer(surface));
		assertWithinSettlementVicinity("in a settlement vicinity", bag);
		
		assertTrue("Transfer to settlement", bag.transfer(settlement));
		assertInsideSettlement("After return", bag, settlement);

	}
```

Line 214 is this: 
> Equipment bag = EquipmentFactory.createEquipment(EquipmentType.BAG, settlement);

**Comment by mokun** [comment]
@bevans2000 

I saw some failed loading tests as follows:

Wonder if the changes you made yesterday on loading/unloading may have somehow affecting that ?!

```java
Error:  Tests run: 12, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 0.221 s <<< FAILURE! -- in com.mars_sim.core.vehicle.task.LoadControllerTest
Error:  com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadFull -- Time elapsed: 0.037 s <<< ERROR!
java.lang.IllegalArgumentException: Cannot store unit type:CONTAINER
	at com.mars_sim.core.UnitManager.addUnit(UnitManager.java:335)
	at com.mars_sim.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:86)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlementEquipment(LoadControllerTest.java:331)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlement(LoadControllerTest.java:490)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoading(LoadControllerTest.java:299)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadFull(LoadControllerTest.java:290)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at junit.framework.TestCase.runBare(TestCase.java:142)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:130)
	at junit.framework.TestSuite.runTest(TestSuite.java:241)
	at junit.framework.TestSuite.run(TestSuite.java:236)

Error:  com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadMissingOptionalEquipment -- Time elapsed: 0.019 s <<< ERROR!
java.lang.IllegalArgumentException: Cannot store unit type:CONTAINER
	at com.mars_sim.core.UnitManager.addUnit(UnitManager.java:335)
	at com.mars_sim.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:86)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlementEquipment(LoadControllerTest.java:331)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlement(LoadControllerTest.java:490)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadMissingOptionalEquipment(LoadControllerTest.java:146)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at junit.framework.TestCase.runBare(TestCase.java:142)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:130)
	at junit.framework.TestSuite.runTest(TestSuite.java:241)
	at junit.framework.TestSuite.run(TestSuite.java:236)

Error:  com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadOptionalEquipment -- Time elapsed: 0.012 s <<< ERROR!
java.lang.IllegalArgumentException: Cannot store unit type:CONTAINER
	at com.mars_sim.core.UnitManager.addUnit(UnitManager.java:335)
	at com.mars_sim.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:86)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlementEquipment(LoadControllerTest.java:331)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlement(LoadControllerTest.java:490)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoading(LoadControllerTest.java:299)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadOptionalEquipment(LoadControllerTest.java:133)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at junit.framework.TestCase.runBare(TestCase.java:142)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:130)
	at junit.framework.TestSuite.runTest(TestSuite.java:241)
	at junit.framework.TestSuite.run(TestSuite.java:236)

Error:  com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadRequiredEquipment -- Time elapsed: 0.011 s <<< ERROR!
java.lang.IllegalArgumentException: Cannot store unit type:CONTAINER
	at com.mars_sim.core.UnitManager.addUnit(UnitManager.java:335)
	at com.mars_sim.core.equipment.EquipmentFactory.createEquipment(EquipmentFactory.java:86)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlementEquipment(LoadControllerTest.java:331)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.loadSettlement(LoadControllerTest.java:490)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoading(LoadControllerTest.java:299)
	at com.mars_sim.core.vehicle.task.LoadControllerTest.testLoadRequiredEquipment(LoadControllerTest.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at junit.framework.TestCase.runTest(TestCase.java:177)
	at junit.framework.TestCase.runBare(TestCase.java:142)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:130)
	at junit.framework.TestSuite.runTest(TestSuite.java:241)
	at junit.framework.TestSuite.run(TestSuite.java:236)
```

**Comment by bevans2000** [comment]
> @bevans2000 
> 
> I saw some failed loading tests as follows:
> 
> Wonder if the changes you made yesterday on loading/unloading may have somehow affecting that ?!
> 
No because this branch does not have those changes. And the PRs for the Loading changes passed all unit tests

**Comment by mokun** [comment]
@muybien3433 

After you add back addDisplayUnit(unit) here under case VEHICLE, can you merge the latest changes from master branch into your branch to see if any failed errors on your PR will automatically go away ?

**Comment by bevans2000** [comment]
There is no need to merge the master I feel unless there is a conflict

**Comment by muybien3433** [comment]
Ye.. couldn't find those maven tests. I'll make it work now.

**Comment by bevans2000** [comment]
@mokun don't worry the Sonar check has failed. This is a problem with the source branch being in a different repo. It should be fixed now on the main branch

**Pull request merged**